src/providers/spatialite/qgsspatialitefeatureiterator.cpp

```diff
@@ -588,6 +588,14 @@ QVariant QgsSpatiaLiteFeatureIterator::getFeatureAttribute( sqlite3_stmt *stmt,
     return sqlite3_column_double( stmt, ic );
   }
 
+  if ( sqlite3_column_type( stmt, ic ) == SQLITE_BLOB )
+  {
+    // BLOB value
+    int blob_size = sqlite3_column_bytes( stmt, ic );
+    const void *blob = sqlite3_column_blob( stmt, ic );
+    return QByteArray( ( const char * ) blob, blob_size );
```

Comment:
```suggestion
    return QByteArray( static_cast<const char *>( blob ), blob_size );
```

Super minor nitpicking, static_casts are the preferred way to handle this. (also in qgsspatialiteprovider.cpp )

Reply:
I totally agree, I actually wrote it like that originally, but I try to follow the style close to the code, and a few lines down there's another C-style cast.

```
    // TEXT value
    const QString txt = QString::fromUtf8( ( const char * ) sqlite3_column_text( stmt, ic ) );
```

I'll follow your suggestion here, and we can perhaps make a separate PR making it consistently C++-style casts elsewhere.